tests/testthat/test-formatters.R

```diff
@@ -537,3 +537,260 @@ mpf <- MatrixPrintForm(strings = strs, spans = spans, aligns = aligns,
+  )
+  bmf$strings[, 4] <- bmf$aligns[, 3] <- sample_list_aligns
+
+  expect_error(cw <- propose_column_widths(bmf), regexp = "*1.1111 | (<0.0001)*")
+  bmf$aligns[nrow(bmf$aligns), c(2, 3)] <- "center"
+  cw <- propose_column_widths(bmf)
+  cw[3] <- cw[3] + 4
+  res_dec <- strsplit(toString(bmf, widths = cw, hsep = "-"), "\\n")[[1]]
+
+  expected <- c(
+    "           mpg           wt                         left   ",
+    "-----------------------------------------------------------",
+    "a          11                     11              decimal  ",
+    "b          11.           11.                      left     ",
+    "c          11.1                            11.1   right    ",
+    "d          11.11                        11.11     dec_right",
+    "e          11.111                11.111           center   ",
+    "f          11.1111          11.1111               dec_left ",
+    "g          11%                              11%   right    ",
+    "h          11.%                   11.%            center   ",
+    "i          11.1%                 11.1%            center   ",
+    "j          11.11%        11.11%                   left     ",
+    "k          11.111%                      11.111%   dec_right",
+    "l       (N=11)                       (N=11)       dec_right",
+    "m        >999.9                          >999.9   right    ",
+    "n        >999.99          >999.99                 dec_left ",
+    "o   1.1111 | (<0.0001)     1.1111 | (<0.0001)     dec_left "
+  )
+  expect_identical(res_dec, expected)
+})
+
+test_that("All 2d cases for decimal alignment", {
+  formats2d <- list_valid_format_labels()$`2d`
+  hard_c <- sapply(formats2d, gsub, pattern = "x", replacement = "1")
+  reduced_df <- mtcars[seq_along(hard_c), c(1, 6)]
+  rownames(reduced_df) <- formats2d
+  colnames(reduced_df) <- c("dec_left", "decimal")
+  bmf <- basic_matrix_form(reduced_df)
+  bmf$strings[seq(2, length(hard_c) + 1), seq(2, 3)] <- as.character(hard_c)
+  bmf$formats[seq(2, length(hard_c) + 1), seq(2, 3)] <- hard_c
+
+  bmf$aligns[, 2] <- "decimal"
+  bmf$aligns[, 3] <- "dec_right"
+  bmf$col_widths <- NULL
+  expect_error(res_dec <- strsplit(toString(bmf, hsep = "-"), "\\n")[[1]],
+               regexp = "*first 3 selected from column dec_left*")
+
+  # expected <- c(
+  #   "                          dec_left                     decimal",
+  #   "--------------------------------------------------------------",
+  #   "xx / xx                11 / 11               11 / 11          ",
+  #   "xx. / xx.              11. / 11.             11. / 11.        ",
+  #   "xx.x / xx.x            11.1 / 11.1           11.1 / 11.1      ",
+  #   "xx.xx / xx.xx          11.11 / 11.11         11.11 / 11.11    ",
+  #   "xx.xxx / xx.xxx        11.111 / 11.111       11.111 / 11.111  ",
+  #   "N=xx (xx%)           N=11 (11%)            N=11 (11%)         ",
+  #   "xx (xx%)               11 (11%)              11 (11%)         ",
```

Comment:
I will take this out before merging but it is here to show how it works for multiple numbers (at the moment it would only select the first one to align). Still, if one day this will be necessary, the algorithm is already "aware" of the possibility of multiple numbers in the same cell